code/modules/multiz/mobile_ladder.dm

```diff
@@ -0,0 +1,104 @@
+	slot_flags = SLOT_BACK
+
+/obj/item/mobile_ladder/proc/place_ladder(atom/A, mob/user)
+	if(isopenspace(A)) //Place into open space
+		var/turf/below_loc = GetBelow(A)
+		if (!below_loc || (istype(/turf/space, below_loc)))
+			to_chat(user, SPAN_NOTICE("Why would you do that?! There is only infinite space there..."))
+			return
+		user.visible_message(SPAN_WARNING("[user] begins to lower \the [src] into \the [A]."),
+			SPAN_WARNING("You begin to lower \the [src] into \the [A]."))
+		if (!handle_action(A, user))
+			return
+
+		var/obj/structure/ladder/mobile/downer  = new(below_loc)
+		var/obj/structure/ladder/mobile/current = new(A)
+		downer.target_up = current
+		current.target_down = downer
+
+		transfer_fingerprints_to(downer)
+		transfer_fingerprints_to(curent)
+		user.drop_from_inventory(src,get_turf(src))
+		qdel(src)
+
+	else if (istype(A, /turf/simulated/floor) || istype(A, /turf/unsimulated/floor))	//Place onto Floor
+		var/turf/upper_loc = GetAbove(A)
+		if (!upper_loc || !isopenspace(upper_loc))
+			to_chat(user, SPAN_NOTICE("There is something above. You can't deploy!"))
+			return
+		user.visible_message(SPAN_WARNING("[user] begins deploying \the [src] on \the [A]."),
+			SPAN_WARNING("You begin to deploy \the [src] on \the [A]."))
+		if (!handle_action(A, user))
+			return
+
+		var/obj/structure/ladder/mobile/current = new(A)
+		var/obj/structure/ladder/mobile/upper   = new(upper_loc)
+		current.target_up = upper
+		upper.target_down = current
+
+		transfer_fingerprints_to(current)
+		transfer_fingerprints_to(upper)
+		user.drop_from_inventory(src,get_turf(src))
+		qdel(src)
+
+/obj/item/mobile_ladder/afterattack(atom/A, mob/user,proximity)
+	if (!proximity)
+		return
+
+	place_ladder(A,user)
+
+/obj/item/mobile_ladder/proc/handle_action(atom/A, mob/user)
+	if (!do_after(user, 30, user))
+		to_chat(user, SPAN_WARNING("Can't place ladder! You were interrupted!"))
+		return FALSE
+	if (!A || QDELETED(src) || QDELETED(user))
+		// Shit was deleted during delay, call is no longer valid.
+		return FALSE
+	return TRUE
+
+/obj/structure/ladder/mobile
+	base_icon = "mobile_ladder"
+	climb_time = 4 SECONDS
+	draw_shadow = FALSE
+	tool_interaction_flags = 0
+
+/obj/structure/ladder/mobile/verb/fold()
+	set name = "Fold Ladder"
+	set category = "Object"
+	set src in oview(1)
+
+	if(usr.incapacitated() || !usr.check_dexterity(DEXTERITY_GRIP, silent = TRUE) || !ishuman(usr))
+		to_chat(usr, SPAN_WARNING("You can't do that right now!"))
+		return
+
+	var/mob/living/carbon/human/H = usr
+	H.visible_message(SPAN_NOTICE("[H] starts folding up [src]."),
+		SPAN_NOTICE("You start folding up [src]."))
+
+	if(!do_after(H, 30, src))
+		to_chat(H, SPAN_WARNING("You are interrupted!"))
+		return
+
+	if(QDELETED(src))
+		return
+
+	var/obj/item/mobile_ladder/R = new(get_turf(H))
+	transfer_fingerprints_to(R)
+
+	H.visible_message(SPAN_NOTICE("[H] folds [src] up into [R]!"),
+		SPAN_NOTICE("You fold [src] up into [R]!"))
+
+	if(target_down)
```

Comment:
Will need to typecheck target_down/target_up as the ladder code doesn't care about type and will happily connect these ones to existing ladders.